src/geom/rt/ray.rs

```diff
@@ -53,12 +53,12 @@ impl Ray {
             Vec3::y_axis()
         };
 
-        let pitch_axis = self.dir.cross(&arbitrary_axis.into());
+        let pitch_axis = self.dir.cross_vec(&arbitrary_axis.into());
```

Comment:
What is the difference between `cross`, `cross_vec` and `cross_dir`?

Is it generally better to use the specialised `cross_vec`/`cross_dir` when the argument is known to be strictly either a `Vec<N>`/`Dir<N>`? - I'd guess they can make internal optimisations when the argument type is concretely known